### ISSUE TITLE
Conflict:  TypeCode exists in both Xamarin.Auth and System.Runtime

### ISSUE THREAD

**Issue by XzaR90** [xamarin/Xamarin.Auth#221]
# Xamarin.Auth Issue

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0433	The type 'TypeCode' exists in both 'Xamarin.Auth, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null' and 'System.Runtime, Version=4.0.20.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

### Version

{
  "dependencies": {
    "Fody": "2.1.2",
    "FreshMvvm": "2.2.0",
    "Microsoft.NETCore.UniversalWindowsPlatform": "5.4.0",
    "Newtonsoft.Json": "10.0.3",
    "PCLCrypto": "2.0.147",
    "PInvoke.BCrypt": "0.5.97",
    "PInvoke.Kernel32": "0.5.97",
    "PInvoke.NCrypt": "0.5.97",
    "PInvoke.Windows.Core": "0.5.97",
    "PropertyChanged.Fody": "2.1.4",
    "Validation": "2.4.15",
    "Xam.Plugin.Connectivity": "3.0.2",
    "Xamarin.Auth": "1.5.0.3",
    "Xamarin.Forms": "2.3.4.267"
  },
  "frameworks": {
    "uap10.0.10586": {}
  },
  "runtimes": {
    "win10-arm": {},
    "win10-arm-aot": {},
    "win10-x86": {},
    "win10-x86-aot": {},
    "win10-x64": {},
    "win10-x64-aot": {}
  }
}

### Steps to reproduce

Using the enum System.TypeCode anywhere in shared project.

### Expected behaviour

To build successfully.

### Actual behaviour

Cannot build.

### Code:

```
            switch (Convert.GetTypeCode(type))
            {
                case TypeCode.Int32:
                    return PropertyBaseModel.INT32;
                case TypeCode.Int64:
                    return PropertyBaseModel.INT64;
                case TypeCode.Single:
                    return PropertyBaseModel.FLOAT;
                case TypeCode.Double:
                    return PropertyBaseModel.DOUBLE;
                case TypeCode.Decimal:
                    return PropertyBaseModel.DECIMAL;
                case TypeCode.Boolean:
                    return PropertyBaseModel.BOOL;
                case TypeCode.DateTime:
                    return PropertyBaseModel.DATETIME;
                case TypeCode.String:
                    return PropertyBaseModel.STRING;
                default:
                    return PropertyBaseModel.NULL;
            }
```

**Comment by moljac** [comment]
@XzaR90 
I'll need minimal repro sample from you. I have 3 UWP samples right now (Traditional and 2 Xamarin.Forms) and cannot reproduce.

There must be something else involved.

TypeCode had to be added 2 years ago when a lot of stuff was missing and I was adding Windows platforms.

**Comment by zeko78** [comment]
@moljac You are not able to reproduce it in UWP samples `becausesource/Core/Xamarin.Auth.Portable/System/TypeCode.cs` does not define TypeCode in UWP because of `#if !WINDOWS_UWP`

I got the same issue when using Xamarin.Auth 1.6 in a project which references a shared project using System.TypeCode from .NET standard 2.0.

System.TypeCode should not be defined when compiling against .NET standard

**Comment by zeko78** [comment]
@moljac Ok, solved the issue at last.

Adding Xamarin.Auth 1.6 to the project that references Xamarin.Auth.XamarinForms 1.6 solved the issue. If Xamarin.Auth 1.6 is not referenced in the project, version 1.5 will be used (as it's the minimum dependency version for Xamarin.Auth.XamarinForms 1.6), and v1.5 does not work nicely with .NET Standard 2.0.

So the fix should be to update the minimum dependency version of Xamarin.Auth to 1.6 in Xamarin.Auth.XamarinForms 1.6

Oh, how I miss the simpler days of dll-hell :-)

**Comment by newky2k** [comment]
resolved with update to 1.6
